firmware/Makefile

```diff
@@ -44,3 +45,9 @@ endif
 
 st-flash: build
 	st-flash --reset write $(BUILD_DIR)/$(PROJECT_NAME).bin 0x08000000
+
+debug-openocd:
```

Comment:
Is there a way to run the openocd and gdb commands simultaneously so you don't need to open a second terminal? Something like https://phoenixnap.com/kb/linux-run-command-background

Need to check if that works on Windows.

Reply:
It is possible but running openocd in the background of gdb will cause issues when the board is unplugged. Openocd will infinitely poll for the connection and there is no way to configure openocd to turn polling off. Even if the terminal window is closed, other terminal windows trying openocd will conflict with the first instance since it is running in the background. The only way to stop it is executing `pkill openocd` (not very user friendly).

We can keep openocd/gdb in separate commands or run simultaneously with the `pkill openocd` note. Explored solutions to kill openocd on disconnection but they are too complex/involved.

Will defer to your opinion on this.

Reply:
Didn't know about that error. Let's keep them separate to avoid that problem